plugin/pkg/shoot/validator/admission.go

```diff
@@ -936,9 +936,12 @@ func getMachineImage(machineImages []garden.MachineImage) (*garden.MachineImage,
 	if len(machineImages) == 0 {
 		return nil, errors.New("the cloud profile does not contain any machine image - cannot create shoot cluster")
 	}
-	// If the shoot does not specify a machine image then we consider the first one specified in the cloud profile to be the
-	// default machine image and return it.
-	return &machineImages[0], nil
+	// If the shoot does not specify a machine image then we determine the latest version in the cloudProfile and consider it as the default
+	_, latestMachineImage, err := helper.DetermineLatestMachineImage(machineImages)
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine latest machine image from cloud profile: %s", err.Error())
+	}
+	return &latestMachineImage, nil
 }
 
 func validateMachineImagesConstraints(constraints []garden.MachineImage, image, oldImage *garden.MachineImage) (bool, []string) {
```

Comment:
We could also validate that no expired image is used here. WDYT?

Reply:
I think it makes sense: this way we can be sure that all shoot clusters have a certain machine image version two days (MaintenanceTimeFrame + defer for one day)  after the expiration date and can potentially remove the image from the CloudProfile.

Reply:
Exactly! Please make sure that no shoot can be created or updated with a machine image having an expired expiration date. It should still be possible to create or update shoots with a machine having having an expiration date that is in the future/not yet expired.

Additionally, I thought about ensuring via another admission plugin that a machine image with an expiration date that is still used by a shoot cannot be removed from a `CloudProfile`. However, I don't think it's needed. We should allow that an administrator removes such a machine image with an expiration date that is in the future. To support this, I think your code in the maintenance controller needs to be updated: If the machine image version currently used by a shoot cannot be found then we also want to update it IMMEDIATELY to the latest version.

Reply:
_We should allow that an administrator removes such a machine image with an expiration date that is in the future_

Do you mean _We should allow that an administrator removes such a machine image with an expiration date that is **in the past**_?

With IMMEDIATELY do you still mean within the maintenance time window?

Then an admin can delete a machineImage that has an expiration date in the past (while making sure it is not the only one in the cloud profile with this machineImage.Name) -> in the next maintenance time window all shoots who's image is not in the cloud profile  get updated to latest machine image version.

Reply:
> Do you mean We should allow that an administrator removes such a machine image with an expiration date that is in the past?

No, in the future.

> With IMMEDIATELY do you still mean within the maintenance time window?

Yes, sure. The code will be in the maintenance controller, hence, only executed during the shoot's maintenance time window.